src/main/java/com/glodblock/github/network/wrapper/FCNetworkWrapper.java

```diff
@@ -0,0 +1,238 @@
+ * a series of unique message types with each having a unique handler. Generally, context should be derived at message
+ * handling time.
+ * <p>
+ * Usage is simple:
+ * <ul>
+ * <li>construct, and store, an instance of this class. It will automatically register and configure your underlying
+ * netty channel.
+ *
+ * <li>Then, call {@link #registerMessage(Class, Class, byte, Side)} for each message type you want to exchange
+ * providing an {@link IMessageHandler} implementation class as well as an {@link IMessage} implementation class. The
+ * side parameter to that method indicates which side (server or client) the <em>message processing</em> will occur on.
+ * The discriminator byte should be unique for this channelName - it is used to discriminate between different types of
+ * message that might occur on this channel (a simple form of message channel multiplexing, if you will).
+ * <li>To get a packet suitable for presenting to the rest of minecraft, you can call {@link #getPacketFrom(IMessage)}.
+ * The return result is suitable for returning from things like {@link TileEntity#getDescriptionPacket()} for example.
+ * <li>Finally, use the sendXXX to send unsolicited messages to various classes of recipients.
+ * </ul>
+ * <p>
+ * Example <code>
+ * <pre>
+ *  // Request message
+ *  public Message1 implements IMessage {
+ *  // message structure
+ *   public fromBytes(ByteBuf buf) {
+ *    // build message from byte array
+ *   }
+ *   public toBytes(ByteBuf buf) {
+ *    // put message content into byte array
+ *   }
+ *  }
+ *  // Reply message
+ *  public Message2 implements IMessage {
+ *   // stuff as before
+ *  }
+ *  // Message1Handler expects input of type Message1 and returns type Message2
+ *  public Message1Handler implements IMessageHandler<Message1,Message2> {
+ *   public Message2 onMessage(Message1 message, MessageContext ctx) {
+ *    // do something and generate reply message
+ *    return aMessage2Object;
+ *   }
+ *  }
+ *  // Message2Handler expects input of type Message2 and returns no message (IMessage)
+ *  public Message2Handler implements IMessageHandler<Message2,IMessage> {
+ *   public IMessage onMessage(Message2 message, MessageContext ctx) {
+ *    // handle the message 2 response message at the other end
+ *    // no reply for this message - return null
+ *    return null;
+ *   }
+ *  }
+ *
+ *  // Code in a {@link FMLPreInitializationEvent} or {@link FMLInitializationEvent} handler
+ *  SimpleNetworkWrapper wrapper = NetworkRegistry.newSimpleChannel("MYCHANNEL");
+ *  // Message1 is handled by the Message1Handler class, it has discriminator id 1 and it's on the client
+ *  wrapper.registerMessage(Message1Handler.class, Message1.class, 1, Side.CLIENT);
+ *  // Message2 is handled by the Message2Handler class, it has discriminator id 2 and it's on the server
+ *  wrapper.registerMessage(Message2Handler.class, Message2.class, 2, Side.SERVER);
+ *  </pre>
+ * </code>
+ *
+ * @author cpw
+ */
```

Comment:
I think describing why this class is needed would be better than copying the entire Javadoc from `SimpleNetworkWrapper`